internal/provider/data_source_gitlab_user_keys.go

```diff
@@ -0,0 +1,70 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_user_keys", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_user_keys`" + ` data source allows a list of SSH keys to be retrieved by either the user ID or username.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/users.html#list-ssh-keys-for-user)`,
+
+		ReadContext: dataSourceGitlabUserKeysRead,
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Description: "The user ID.",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
+			"keys": {
+				Description: "The user's keys.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: datasourceSchemaFromResourceSchema(gitlabUserSSHKeySchema(), nil, nil),
+				},
+			},
+		},
+	}
+})
+
+func dataSourceGitlabUserKeysRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	var keys []*gitlab.SSHKey
+	var err error
+
+	log.Printf("[INFO] Reading Gitlab user")
+
+	userIDData, userIDOk := d.GetOk("user_id")
+
+	if userIDOk {
+		// Get SSH keys by user
+		keys, _, err = client.Users.ListSSHKeysForUser(userIDData.(int), &gitlab.ListSSHKeysForUserOptions{}, gitlab.WithContext(ctx))
```

Comment:
🚧 **Suggestion:** we need to implement pagination here, see:

https://github.com/xanzy/go-gitlab/blob/6da2d13f2241eab2edf52f881b28a31595780a7e/users.go#L431

Here is an example of how it's implemented in another data source:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/6540e190e384a86c4fc61da32f86332db1b9d175/internal/provider/data_source_gitlab_cluster_agents.go#L50

Reply:
I've implemented pagination.